### DISCUSSION TITLE
HOW TO ENSURE THAT A PAUSE IN THE VOICE DOESN'T LEAD TO AN BREAKING UP OF MSG..

### DISCUSSION THREAD

**Discussion by DIVANSHMAHAJAN** [speechly/speechly#141] [Q&A]
https://codepen.io/speechly/pen/VwzoMrW?editors=1010
In this also while speaking if we give a pause then previous spoken words get erased off.HOW TO STOP THIS FROM HAPPENING?

CODE:
{segment ?(
<>
{segment.words.map((w)=>w.value).join(" ")}
</>
) : null}

HOW TO GET AROUND THIS? till the time the button is pressed I want to hear the full msg..
whereas https://demos.speechly.com/speech-to-text/ in this it doesnt happen so..

CAN U PROVIDE GITHUB LINK OF THE DEMO CHAT APPLICATION ?

**Comment by aukkonen** [comment]
Hello @DIVANSHMAHAJAN 

You can find code for the demo chat application here: https://github.com/speechly/speechly/tree/main/demos/speech-to-text

However, the demo app explicitly disables "silence triggered segmentation" that you are now observing. Enabling and disabling some of these special parameters is not yet available on the dashboard. To do this, you should use our Command Line Tool to configure and deploy the application. Instructions for this are here: https://docs.speechly.com/dev-tools/command-line-client/

The configuration to deploy with the CLI is in the file `config/config.yaml` in the above repo.

(We are going to release a new version of the dashboard soon that will also allow disabling silence triggered segmentation from there. But this might still take some time.)

Alternatively, if you don't want to use the CLI and want to go on right away, you can work around the issue by collecting the transcript from `segment.words` to another list (or string), and display this other list in your UI instead. This may require a bit of bookkeeping as you only want to append each word from `segment.words` once to this other list, and the callback might be called several times for each new word. Note that right before the `segment.words` list is cleared (when there's a pause) , the segment object has `segment.isFinal == true`, which you can use to determine that there is a new segment starting.
